### ISSUE TITLE
ProcessManager's kill_children function can't reach signal.SIGKILL

### ISSUE THREAD

**Issue by tyeapple** [saltstack/salt#38222]
### Description of Issue/Question
ProcessManager's wait_for_kill(5s) is longer than delay(3s) in /etc/init.d/salt-master, that cause force kill undead children never reached. The parent proc was killed by killproc, left zombies behind.

### Setup
ConnectedCache is hard to stop, when I set "con_cache: True"  in /etc/salt/master, using "service salt-master stop", "ConnectedCache" delayed the closing process. 
```
master_id: m0
order_masters: True
include: master.d/*
user: root
minion_data_cache: True
hash_type: sha256
con_cache: True
presence_events: True
worker_threads: 2
gather_job_timeout: 120
timeout: 60
```

### Steps to Reproduce Issue
master log, after run "service salt-master stop"
```
2016-12-13 11:06:31,346 [salt.utils.process                                   ][687][DEBUG   ][1870] Maintenance received a SIGTERM. Exiting
2016-12-13 11:06:31,347 [salt.utils.process                       ][687][DEBUG   ][1866] EventPublisher received a SIGTERM. Exiting
2016-12-13 11:06:31,349 [salt.transport.zeromq                    ][647][DEBUG   ][1928] ZeroMQReqServerChannel received a SIGTERM. Exiting
2016-12-13 11:06:31,349 [salt.transport.zeromq                    ][505][INFO    ][1928] MWorkerQueue under PID 1928 is closing
2016-12-13 11:06:31,352 [salt.utils.process                       ][687][DEBUG   ][1869] Engine received a SIGTERM. Exiting
2016-12-13 11:06:31,355 [salt.transport.zeromq                                ][505][INFO    ][1932] MWorkerQueue under PID 1932 is closing
2016-12-13 11:06:31,355 [salt.transport.zeromq                                ][505][INFO    ][1933] MWorkerQueue under PID 1933 is closing
2016-12-13 11:06:31,356 [salt.utils.process                                   ][687][DEBUG   ][1932] MWorker received a SIGTERM. Exiting
2016-12-13 11:06:31,359 [salt.utils.process                                   ][687][DEBUG   ][1933] MWorker received a SIGTERM. Exiting
2016-12-13 11:06:31,369 [salt.utils.process                       ][687][DEBUG   ][1927] ReqServer received a SIGTERM. Exiting
```

process stats
```
#ps axuf|grep salt-master| grep -v grep
root      1861  0.7  1.6 518848 33772 ?        Sl   11:06   0:00 /usr/bin/python2.6 /usr/bin/salt-master -d ProcessManager
root      1862  0.0  1.5 429512 30844 ?        S    11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d MultiprocessingLoggingQueue
root      1863  0.0  1.5 442388 32444 ?        Sl   11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d ZeroMQPubServerChannel
root      1866  0.0  1.5 434452 32112 ?        S    11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d EventPublisher
root      1869  0.0  1.6 433988 32892 ?        S    11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d
root      1870  2.0  1.7 435820 35140 ?        S    11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d Maintenance
root      1874  0.0  1.5 453312 32724 ?        Sl   11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d
root      1927  0.0  1.5 518848 32388 ?        S    11:06   0:00  \_ /usr/bin/python2.6 /usr/bin/salt-master -d ReqServer_ProcessManager
root      1928  0.0  1.5 518848 32848 ?        Sl   11:06   0:00      \_ /usr/bin/python2.6 /usr/bin/salt-master -d MWorkerQueue
root      1932  7.1  1.8 534296 38428 ?        Sl   11:06   0:01      \_ /usr/bin/python2.6 /usr/bin/salt-master -d MWorker-0
root      1933  7.2  1.8 534280 38416 ?        Sl   11:06   0:01      \_ /usr/bin/python2.6 /usr/bin/salt-master -d MWorker-1

#service salt-master stop
Stopping salt-master daemon:                               [  OK  ]

#ps axuf|grep salt-master| grep -v grep
root      1862  0.0  1.5 429512 30844 ?        S    11:06   0:00 /usr/bin/python2.6 /usr/bin/salt-master -d MultiprocessingLoggingQueue
root      1874  0.0  1.5 453312 32748 ?        Sl   11:06   0:00 /usr/bin/python2.6 /usr/bin/salt-master -d
```

### Versions Report
#salt --versions-report
Salt Version:
           Salt: 2016.11.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, May 29 2014, 05:49:27)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.2
        machine: x86_64
         system: Linux

**Comment by Ch3LL** [comment]
@dmurphy18 I think this might in your realm and I am not sure on this. Any ideas on this one?

**Comment by tyeapple** [comment]
/etc/init.d/salt-master 
change "killproc $PROCESS" to "killproc -d 10 $PROCESS" will resolve the issue.

**Comment by dmurphy18** [comment]
This will be fixed in the next release

**Comment by dmurphy18** [comment]
Fixed in Salt 21016.3..6, leaving here as a reminder for Salt 2016.11.4 when it is built and packaged

**Comment by dmurphy18** [comment]
Fixed in Salt 2016.11.4
